src/components/FeatureViews/Courses/AttendanceTable.js

```diff
@@ -0,0 +1,437 @@
+  };
+
+  const attendanceStatusUnselectColor = {
+    PRESENT: "#C9FFD5",
+    TARDY: "#FFF6D4",
+    ABSENT: "#FFD8D8",
+  };
+
+  const renderAttendanceStatus = (
+    studentAttendanceList,
+    attendanceListIndex,
+    sessionId
+  ) => {
+    const renderEachButton = ["PRESENT", "TARDY", "ABSENT"].indexOf(studentAttendanceList[attendanceListIndex][sessionId]) >= 0
+    if (renderEachButton) {
+      return (
+        <Button
+          style={{
+            backgroundColor: attendanceStatusSelectColor[studentAttendanceList[attendanceListIndex][studentAttendanceList[attendanceListIndex].sessionId]],
+            color: "black"
+          }}
+          className={classes.buttonStatusStyle}
+          disabled
+        >
+          {formatAttendanceStatusToPascalCase(studentAttendanceList[attendanceListIndex][studentAttendanceList[attendanceListIndex].sessionId])}
+        </Button>
+      );
+    } else {
+      return null;
+    }
+  };
+
+  const renderButtonSelection = (studentIndex, sessionColumn, value, borderTop, borderBottom) => (                      
+  <IconButton
+    data-studentIndex={studentIndex}
+    data-sessionId={sessionColumn.sessionId}
+    data-attendanceId={sessionColumn.attendanceId}
+    value={value}
+    size='small'
+    className={classes.buttonGroupStyle}
+    onClick={handleClick}
+    style={{
+      backgroundColor: `${
+        sessionColumn[sessionColumn.sessionId] === value || sessionColumn[sessionColumn.sessionId] === "UNSET"
+          ? attendanceStatusSelectColor[sessionColumn[sessionColumn.sessionId]]
+          : attendanceStatusUnselectColor[sessionColumn[sessionColumn.sessionId]]
+      }`,
+      ...borderTop,
+      ...borderBottom
+    }}
+  />)
+
+  const sortDescOrder = (firstEl, secondEl) => (firstEl < secondEl ? -1 : 0);
+  const sortAscOrder = (firstEl, secondEl) => (firstEl > secondEl ? -1 : 0);
+
+  const studentAttendanceDisplay = courseAttendanceMatrix
+    .filter((student) =>
+      sortByAlphabet === "asc" ||
+      sortByAlphabet === "desc" ||
+      sortByAlphabet === ""
+        ? student
+        : {
+            [student.studentName]: student.studentName,
+          }[sortByAlphabet]
+    )
+    .sort(
+      (firstEl, secondEl) =>
+        ({
+          asc: sortAscOrder(firstEl.studentName, secondEl.studentName),
+          desc: sortDescOrder(firstEl.studentName, secondEl.studentName),
+        }[sortByAlphabet])
+    );
+
+  return (
+    <TableContainer className={classes.table}>
+      <Table aria-label='simple table'>
+        <TableHead>
+          <TableRow>
+            <TableCell style={{ color: "black" }}>
+              <StudentFilterOrSortDropdown
+                students={studentsFullNameList}
+                setSortByAlphabet={setSortByAlphabet}
+                sortByAlphabet={sortByAlphabet}
+              />
+            </TableCell>
+            {sessions
+              .sort((firstSession, secondSession) => firstSession.id - secondSession.id)
```

Comment:
pull sort function into an explanatory variable. sortByRecentSessions ? not too sure